### PR TITLE
Change format of zero types

### PR DESCRIPTION
**Pull request by yassinebenaid**
What is this PR?
= It changes the format of nil maps, slices,  functions, pointers and channels

What was changed?
-  Slices:

    ```go
    var x []int
    var y = make([]int, 0)
    ```
    Before:
    ```txt
    []int:0:0 {}
    []int:0:0 {}
    ```
    Now:
    ```txt
    []int(nil)
    []int:0:0 {}
    ```
-  Maps:

    ```go
    var x map[string]int
    var y = make(map[string]int)
    ```
    Before:
    ```txt
    map[string]int:0 {}
    map[string]int:0 {}
    ```
    Now:
    ```txt
    map[string]int(nil)
    map[string]int:0 {}
    ```
-  Channels:

    ```go
    var x chan string
    var y = make(chan string)
    ```
    Before:
    ```txt
    chan string
    chan string
    ```
    Now:
    ```txt
    chan string(nil)
    chan string
    ```
-  Functions:

    ```go
    var x func()
    var y = func(){}
    ```
    Before:
    ```txt
    func()
    func()
    ```
    Now:
    ```txt
    func()(nil)
    func()
    ```
-  Pointers:

    ```go
    var x *int
    var y = &x
    ```
    Before:
    ```txt
    nil
    &nil
    ```
    Now:
    ```txt
    *int(nil)
    &*int(nil)
    ```

Why would we need to this?
= Some people like myself are using `godump` in tests for logging, and we compare two data structures using `reflect.DeepEqual()`, When comparing a nil slice with a slice of zero length and capacity, they are not equal, but they look the same in logs. 

Why there are changes in workflows?
= I don't know, I did a git rebase to solve some issues and I think they caused the 3 first commits not to be recognized in master (hash may have changed).

### CODE REVIEW

File: dumper.go [hunk 1]
```diff
@@ -198,8 +198,16 @@ func (d *Dumper) dump(val reflect.Value, ignoreDepth ...bool) {
 		d.dumpMap(val)
 	case reflect.Func:
 		d.buf.WriteString(__(d.Theme.Func, val.Type().String()))
+		if val.IsNil() {
+			d.buf.WriteString(__(d.Theme.Braces, "(") + __(d.Theme.Nil, "nil") + __(d.Theme.Braces, ")"))
```
Comment (ccoVeille) [inline dumper.go:202]: Based on the fact you are repeating the sane pattern multiple times, can't it be refactored to define one functional and call it everywhere?

Comment (yassinebenaid) [reply dumper.go:202]: I thought it would be good to keep it verbose, But, I'll refa tor that.

Comment (yassinebenaid) [reply dumper.go:202]: I don't like to create colored functions that do one simple operation.

So, the best option I have on my head is to create a private property that holds that string and keep using it everywhere.

**Review by ccoVeille — DISMISSED** [review]
I found one last typo,but then it's OK 👍

**Pull request merged**